Widget/Fitness/FitnessWidget.swift

```diff
@@ -0,0 +1,238 @@
+}
+
+private func getBusyString(room: FitnessRoom) -> String {
+    let hours = getHours(room: room)
+    let date = Date()
+    if date < hours.0 || date > hours.1 {
+        return ""
+    } else if room.capacity == 0.0 {
+        return "Empty"
+    } else if room.capacity < 10.0 {
+        return "Not very busy"
+    } else if room.capacity < 30.0 {
+        return "Slightly busy"
+    } else if room.capacity < 60.0 {
+        return "Pretty busy"
+    } else if room.capacity < 90.0 {
+        return "Extremely busy"
+    } else {
+        return "Packed"
+    }
+}
+
+private func getOpenString(room: FitnessRoom, showbusystring: Bool) -> String {
+    let hours = getHours(room: room)
+    let date = Date()
+    let dateFormatter = DateFormatter()
+    dateFormatter.dateFormat = "h:mm a"
+
+    if(date < hours.0) {
+        return "Closed • Opens at " + dateFormatter.string(from: hours.0).replacingOccurrences(of: ":00", with: "")
+    } else if (hours.0 <= date && date <= hours.1) {
+        return "Open until " + dateFormatter.string(from: hours.1).replacingOccurrences(of: ":00", with: "") + ((showbusystring) ? (" • " + getBusyString(room: room)) : "")
+    } else {
+        return "Closed"
+    }
+}
+
+private func getHours (room: FitnessRoom) -> (Date, Date) {
+    let weekdayIndex: Int = (Calendar.current.component(.weekday, from: Date()) + 5) % 7
+    var hours: (Date, Date) {
+        let calendar = Calendar.current
+        let currentDate = Date()
+        let timeFormatter = DateFormatter()
+        timeFormatter.dateFormat = "HH:mm:ss"
+        let openTime = timeFormatter.date(from: room.open[weekdayIndex])!
+        let closeTime = timeFormatter.date(from: room.close[weekdayIndex])!
+
+        let openDate = calendar.date(bySettingHour: openTime.hour, minute: openTime.minutes, second: 0, of: currentDate)!
+        let closeDate = calendar.date(bySettingHour: closeTime.hour, minute: closeTime.minutes, second: 0, of: currentDate)!
+        return (openDate, closeDate)
+    }
+    return hours
+}
+
+private func isOpen (room: FitnessRoom) -> Bool {
+    let date = Date()
+    let hours = getHours(room: room)
+    return hours.0 <= date && date <= hours.1
+}
+
+
+
+private struct FitnessWidgetSmallView: View {
+    var rooms: [FitnessRoom]
+    var configuration: ConfigureFitnessWidgetIntent.Configuration
+    
+    
+    var body: some View {
+        VStack {
+            Text(rooms[0].name)
+                .font(.system(size: 13, weight: .semibold))
+                .padding(.top)
+            Text(getOpenString(room: rooms[0], showbusystring: false))
+                .font(.system(size: 11))
+                .foregroundStyle(isOpen(room: rooms[0]) ? .green : .blue)
+                .multilineTextAlignment(.center)
+                .fixedSize(horizontal: false, vertical: true)
+            MeterView(current: isOpen(room: rooms[0]) ? rooms[0].capacity : 0, maximum: 100.0, style: Color.blue, lineWidth: 6) {
```

Comment:
Can we make the meter color `Color.primary` if the background's `prefersGrayscaleContent` is true?